client/signup/steps/survey/survey-step-v2.jsx

```diff
@@ -29,8 +32,7 @@ export default React.createClass( {
 
 	getInitialState() {
 		return {
-			shouldShowOther: false,
-			otherWriteIn: '',
+			otherWriteIn: get( find( this.props.signupProgressStore, { stepName: this.props.stepName } ), 'otherWriteIn', '' ),
```

Comment:
It's an anti-pattern to use [props in `getInitialState`](https://facebook.github.io/react/tips/props-in-getInitialState-as-anti-pattern.html). Could we abstract this to a getter method:

```
getOtherWriteIn() {
  return this.state.otherWriteIn || get( find( this.props.signupProgressStore, { stepName: this.props.stepName } ), 'otherWriteIn', '' );
}
```

props @sirbrillig for making me look smart ;)